### ISSUE TITLE
Implementation of Efficient-Net and Efficient-Det in Pytorch Lightning

### ISSUE THREAD

**Issue by bilel-bj** [PyTorchLightning/lightning-bolts#1]
## 🚀 Feature
Implementation of Efficient-Net and Efficient-Det in Pytorch Lightning

### Motivation

These two algorithms are the state of the art in Image classification and Object detection. Providing an implementation in Pytorch LIghtning will attract many data scientist to adopt this promising framework

### Alternatives

https://paperswithcode.com/paper/efficientdet-scalable-and-efficient-object

**Comment by edgarriba** [comment]
when you say implementation means everything including the network from scratch, or just mixing lightning with e.g torchvision SOTA models ?

**Comment by bilel-bj** [comment]
It is not actually provided in TorchVision. We can only find 8it in Torch Hub. It will be attractive that we find some easy way to use these algorithms in Lightning. Thanks for your support.

**Comment by iqDF** [comment]
Hi guys, is this still on? I can have a helping hand in starting the 6th of May.

**Comment by justusschock** [comment]
I think so far, there is no-one working on this, so if you want to, we'd appreciate that :)

**Comment by iqDF** [comment]
Will do that, but gimme some time. I'm on another finishing project. I'll let get back here asap I promise.

**Comment by iqDF** [comment]
Hi, so regarding the migration from pytorch to pytorch lightning for Efficient Net:
https://paperswithcode.com/paper/efficientdet-scalable-and-efficient-object

Do you guys have any requirements or certain standard for the code? I recon that the work here is intended for example rather than solely for implementation so I think there is a need for self-documenting code that's easy to digest perhaps? 

Please list, add, edit or delete the ideas that deemed necessary/uneccessary. I'll start working on it this week - 25th of May

List of requirements:

- [ ] Working and reproducible Efficient Net using Pytorch Lightning
   - Work on CPU and GPU
   - Work on Pytorch 1.4
   - Work on Latest Pytorch Lightning  
- [ ] Perform benchmarking as sanity check for accuracy and reproducibility on original work
- [ ] Checkpoints models / pretrained weights on the said dataset

**Comment by justusschock** [comment]
@iqDF Sorry for the late reply. Of course you can ping  me whenever you need :) I recently get spammed with github notifications, so I mussed have missed this one :) As for documentation: Yes, a few doctest examples would be nice. 

And for the hparams, just include a default that matches the hparams you used to achieve the reported results :)

**Comment by Borda** [comment]
@iqDF how is it going here?

**Comment by iqDF** [comment]
@Borda Hi I almost thought this was abandoned due to delay inactivity in the forum. Anyway here is the progress:

1. Currently I have used several detection models (efficient det included) in pytorch  + lightning for my kaggle project.
2. The implementation needs to be tidy up. I use it for competition currently but tidying up should be quick, but I'll take my time for one week.
3. Performance wise, I'll just use pretrained one from https://github.com/google/automl

**Comment by Borda** [comment]
everyone is busy, but seeing this landed would be great, I personally would love to try it...
if you start a draft PR we can help you with some work... :]

**Comment by ydcjeff** [comment]
@williamFalcon @Borda I would like to implement EfficientNet.
Do I just need to implement just model with nn.Module or full train/val/test loop with LightningModule (including DataModule)?

**Comment by JVGD** [comment]
Hi people. What's the state of the EfficientDet implementation? I took a look at this [WIP branch](https://github.com/PyTorchLightning/pytorch-lightning-bolts/pull/31) but it only seems to have the BiFPN module implemented. As you know EfficientDet is EfficientNet + BiFPN... If we can implement this it would attract lots of users since EfficientDets are still SOTA in object detection while being very lightweight to fit into edge devices.

**Comment by akihironitta** [comment]
@oke-aditya I'm sure your PRs like #475 are waiting for our reviews for so long, but let me mention you here since you might be interested in this one.

**Comment by oke-aditya** [comment]
😅 No worries @akihironitta reviewing takes time. 


I think efficient net will be added to torchvision soon. Once efficient net is supported by torchvision adding efficient det will be easier. 

Since efficient net, ResNet, ResNext, mobilenet all are classification models directly from torchvision, it might make sense to add classification trainer ?

There are other recent classification approaches such as DeiT (FB research)
Lambda Networks etc 
So maybe a Classification API is worth adding for bolts ?
